agent/api/task/task.go

```diff
@@ -1513,6 +1544,22 @@ func (task *Task) GetTaskENI() *apieni.ENI {
 	return task.ENI
 }
 
+// SetAppMesh sets the app mesh config of the task
+func (task *Task) SetAppMesh(appMesh *apiappmesh.AppMesh) {
+	task.lock.RLock()
```

Comment:
Here should be task.lock.Lock()/task.lock.Unlock(). RLock is read lock.

Reply:
Didn't know the difference, changed to to regular lock